packages/js/components/src/image-gallery/image-gallery-toolbar.tsx

```diff
@@ -0,0 +1,115 @@
+/**
+ * External dependencies
+ */
+import { createElement } from '@wordpress/element';
+import { Toolbar, ToolbarButton, ToolbarGroup } from '@wordpress/components';
+import { chevronRight, chevronLeft, trash } from '@wordpress/icons';
+import { MediaUpload } from '@wordpress/media-utils';
+
+/**
+ * Internal dependencies
+ */
+import { CoverImageIcon } from './icons';
+import { SortableHandle } from '../sortable';
+import { MediaUploadComponentType } from './types';
+
+export type ImageGalleryToolbarProps = {
+	childIndex: number;
+	moveItem: ( fromIndex: number, toIndex: number ) => void;
+	removeItem: ( removeIndex: number ) => void;
+	replaceItem: (
+		replaceIndex: number,
+		newSrc: string,
+		newAlt: string
+	) => void;
+	setToolBarItem: ( key: string | null ) => void;
+	lastChild: boolean;
+	MediaUploadComponent: MediaUploadComponentType;
+} & React.HTMLAttributes< HTMLDivElement >;
+
+export const ImageGalleryToolbar: React.FC< ImageGalleryToolbarProps > = ( {
+	childIndex,
+	moveItem,
+	removeItem,
+	replaceItem,
+	setToolBarItem,
+	lastChild,
+	MediaUploadComponent = MediaUpload,
+}: ImageGalleryToolbarProps ) => {
+	const moveNext = () => {
+		moveItem( childIndex, childIndex + 1 );
+	};
+
+	const movePrevious = () => {
+		moveItem( childIndex, childIndex - 1 );
+	};
+
+	const setAsCoverImage = ( coverIndex: number ) => {
+		moveItem( coverIndex, 0 );
+		setToolBarItem( null );
+	};
+
+	const isCoverItem = childIndex === 0;
+
+	return (
+		<div className="woocommerce-image-gallery__toolbar">
+			<Toolbar
+				onClick={ ( e ) => e.stopPropagation() }
+				label="Options"
```

Comment:
We'll need to wrap this and the below labels in i18n calls.

```suggestion
				label={ __( 'Options', 'woocommerce' ) }"
```

Reply:
Whoops, forgot that again. Done.